Source/Diagnostics/ReducedDiags/FieldEnergy.cpp

```diff
@@ -142,3 +153,52 @@ void FieldEnergy::ComputeDiags (int step)
      *   ......] */
 }
 // end void FieldEnergy::ComputeDiags
+
+// Function that computes the sum of the field squared in RZ
+amrex::Real
+FieldEnergy::ComputeNorm2RZ(const amrex::MultiFab& field, const int lev)
+{
+    const std::array<amrex::Real,3>& dx = WarpX::CellSize(lev);
+    const amrex::Real dr = dx[0];
+
+    constexpr int NODE = amrex::IndexType::NODE;
+
+    amrex::ReduceOps<amrex::ReduceOpSum> reduce_ops;
+    amrex::ReduceData<amrex::Real> reduce_data(reduce_ops);
+    using ReduceTuple = typename decltype(reduce_data)::Type;
+
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+    for ( amrex::MFIter mfi(field, amrex::TilingIfNotGPU()); mfi.isValid(); ++mfi )
+    {
+
+        amrex::Array4<const amrex::Real> const& field_arr = field.array(mfi);
+
+        amrex::Box tilebox = mfi.tilebox();
+        amrex::Box tb = convert(tilebox, field.ixType().toIntVect());
+
+        // Lower corner of tile box physical domain
+        const std::array<amrex::Real, 3>& xyzmin = WarpX::LowerCorner(tilebox, lev, 0._rt);
+        const Dim3 lo = lbound(tilebox);
+        const Real rmin = xyzmin[0] + (tb.type(0) == NODE ? 0. : 0.5*dx[0]);
```

Comment:
If you want to save the one `NODE` enum, you could write this:
```suggestion
        const Real rmin = xyzmin[0] + (tb.type().nodeCentered(0) ? 0. : 0.5*dx[0]);
```